Sources/NIOHTTP1/HTTPUpgradeHandler.swift

```diff
@@ -74,7 +74,7 @@ public class HTTPServerUpgradeHandler: ChannelInboundHandler, RemovableChannelHa
     private var seenFirstRequest = false
 
     /// The closure that should be invoked when the end of the upgrade request is received if any.
-    private var upgrade: (() -> Void)?
+    private var upgradeState: UpgradeState = .idle
     private var receivedMessages: [NIOAny] = []
```

Comment:
this should be a `CircularBuffer<NIOAny>` because we're looping over `removeFirst()` which is quite expensive as array

Reply:
Good catch, will amend.